doc/auto_storage.md

```diff
@@ -729,6 +781,82 @@ space first by shrinking the partitions and deleting them only if shrinking is n
 }
 ```
 
+### Generating Partitions as MD RAID members
+
+MD arrays can be configured to explicitly use a set of devices by adding their aliases to the
+`devices` property.
+
+```json
+"storage": {
+  "drives": [
+    {
+      "search": "/dev/sda",
+      "partitions": [
+        { "alias": "sda-40", "size": "40 GiB" }
+      ]
+    },
+    {
+      "search": "/dev/sdb",
+      "partitions": [
+        { "alias": "sdb-40", "size": "40 GiB" }
+      ]
+    }
+  ],
+  "mdRaids": [
+    {
+      "devices": [ "sda-40", "sdb-40" ],
+      "level": "raid0"
+    }
+  ]
+}
+```
+
+The partitions acting as members can be automatically generated by simply indicating the target
+disks that will hold the partitions. For that, the `devices` section must contain a `generate`
+entry.
+
+```json
+"storage": {
+  "drives": [
+    { "search": "/dev/sda", "alias": "sda" },
+    { "search": "/dev/sdb", "alias": "sdb" },
+  ],
+  "mdRaids": [
+    {
+      "devices": [
+        {
+          "generate": {
+            "targetDisks": ["sda", "sdb" ],
+            "size": "40 GiB"
+          }
+        }
+      ]
+      "level": "raid0"
+    }
+  ]
+}
+```
+
+As explained at the section about sizes, it's also possible to set the size for the new RAID letting
+Agama calculate the corresponding sizes of the partitions used as members. That allows to use the
+short syntax for `generate`.
+
+```json
+"storage": {
+  "drives": [
+    { "search": "/dev/sda", "alias": "sda" },
+    { "search": "/dev/sdb", "alias": "sdb" },
+  ],
+  "mdRaids": [
+    {
+      "devices": [ { "generate": ["sda", "sdb" ] } ],
+      "level": "raid0",
+      "size": "40 GiB"
```

Comment:
What happens if both *generate* and the raid indicate a size?

Reply:
That makes no much sense, I would say.

Reply:
Ok, but the schema would admit it. So, I guess we will need to check that cases in `ConfigChecker`.